### PR TITLE
[feat] implements basic interest form

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
Creates simple interest form.

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- updated the schema for objects of type `Interest` (added type string for id to facilitate interest inserts)
- Created `insertInterest` and `getAllInterests` queries
- Implemented a basic form that inserts an interest object (with hardcoded data) to the `interests` table in Supabase

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Open the app in local host and navigate to `http://localhost:3000/interest`
- Add dummy data into the textarea input, and press the submit button to complete the action
- Go to the interests table in Supabase and confirm that a new record was added (can look at whether the record count increases)

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
   },
   "dependencies": {
     "@calblueprint/prettier-config": "^0.0.1",
-    "@supabase/supabase-js": "^2.36.0",
+    "@supabase/supabase-js": "^2.37.0",
```
Comment (ccatherinetan) [inline package.json]: Was this just an update in supabase? just checking if there was any significant difference here :)

Comment (varortz) [reply package.json]: This was from when I had to create the auto-generated schema we didn't end up using. I'll revert back to 2.36.0!

**Review by pragyakallanagoudar — APPROVED** [review]
tested it out and it works perfectly!! AMAZINGGG

**Review by pragyakallanagoudar — CHANGES_REQUESTED** [review]
oops just one note - can we discard the `package-lock.json` changes? i think this will cause merge conflicts for other PRs

**Review by pragyakallanagoudar — APPROVED** [review]
NVM - just got rid of them!

**Review by ccatherinetan — APPROVED** [review]
Looks great :D!
